homeassistant/components/roomba/sensor.py

```diff
@@ -0,0 +1,76 @@
+"""Sensor for checking the battery level of Roomba."""
+import logging
+
+from homeassistant.const import DEVICE_CLASS_BATTERY
+from homeassistant.helpers.entity import Entity
+
+from . import roomba_reported_state
+from .const import BLID, DOMAIN, ROOMBA_SESSION
+
+_LOGGER = logging.getLogger(__name__)
+
+
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the iRobot Roomba vacuum cleaner."""
+    domain_data = hass.data[DOMAIN][config_entry.entry_id]
+    roomba = domain_data[ROOMBA_SESSION]
+    blid = domain_data[BLID]
+    roomba_vac = RoombaBattery(roomba, blid)
+    async_add_entities([roomba_vac], True)
+
+
+class RoombaBattery(Entity):
+    """Class to hold Roomba Sensor basic info."""
+
+    def __init__(self, roomba, blid):
+        """Initialize the sensor object."""
+        self.vacuum = roomba
+        self.vacuum_state = roomba_reported_state(roomba)
+        self._blid = blid
+        self._name = self.vacuum_state.get("name")
+        self._identifier = f"roomba_{self._blid}"
+        self._battery_level = None
+
+    @property
+    def name(self):
+        """Return the name of the sensor."""
+        return f"{self._name} Battery Level"
+
+    @property
+    def unique_id(self):
+        """Return the ID of this sensor."""
+        return f"battery_{self._blid}"
+
+    @property
+    def device_class(self):
+        """Return the device class of the sensor."""
+        return DEVICE_CLASS_BATTERY
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit_of_measurement of the device."""
+        return "%"
```

Comment:
```suggestion
        return UNIT_PERCENTAGE
```

Reply:
Need to import it from `from homeassistant.const`

Reply:
the push is done